instrumentation/base/lib/opentelemetry/instrumentation/base.rb

```diff
@@ -369,13 +369,11 @@ def coerce_env_var(env_var, validation_type) # rubocop:disable Metrics/Cyclomati
       end
 
       def wrap_lambda_in_error_handler(lambda_block, option_name)
-        wrapped_lambda = lambda do |*args|
-            lambda_block.call(*args)
-          rescue StandardError => e
-            OpenTelemetry.handle_error(exception: e, message: "Failed to call callable configuration option #{option_name}")
-          end
-
-        wrapped_lambda
+        lambda do |*args|
+          lambda_block.call(*args)
+        rescue StandardError => e
```

Comment:
yea those are good points. happy to close this pr and open a new one with work specific to instrumentation level changes. i'm doing this in response to a comment on slack from @fbogsany . I think what he correctly points out is that we could improve our error handling of callbacks. The specification guidance mentions

> API methods that accept external callbacks MUST handle all errors.

https://opentelemetry.io/docs/specs/otel/error-handling/

Upon reflection, I think we can probably better accomplish this with non-generalized error handling around each invocation of `.call` on a `callable` config option. This would take legwork of going through instrumentation and and wrapping relevant `.call` invocations in a rescue block, allowing us to customize each return value, but that would be on the burden of instrumentation authors and not end users, and seems like a straightforward way to satisfy all requirements here. does that seems like an ok path forward in your eyes?